### ISSUE TITLE
Semver-respecting failSeverity

### ISSUE THREAD

**Issue by nivertius** [revapi/revapi#50]
It would be nice to have special `failSeverity` in maven plugin configuration that would fail the build only if changes to API does not allow current project version according to semver. In other words, current project version would define which actual `failSeverity` is used.

Lets say that the latest release for some project is `1.2.1`, and current snapshot version declared in POM is `1.2.2-SNAPSHOT`. When using this special `failSeverity` this build would fail if there where any non-breaking changes.

But lets say that user skips versions and goes straight to `1.3.0-SNAPSHOT`, for example using `revapi:update-versions`. Then, without editing pom and changing `failSeverity`, the build would pass with non-breaking changes, but still fail for breaking changes. If the version where updated to `2.0.0-SNAPSHOT`, and still using same `failSeverity` the build would always pass.

**Comment by nivertius** [comment]
I understand that this can be achieved throught semver-ignore extension. But still, `failSeverity` is meaningless if this extension is used and could be changed to handle this specific situation.

**Comment by metlos** [comment]
Not all projects follow strict semver rules. I generally opt for extensions instead of parameters of the maven plugin, because the extensions are automatically available in other "modes" of revapi usage, too - in the Ant task and the CLI. 

Note that I'm currently trying to implement #19. Once that's in, the configuration in pom.xml will be all XML which should make things much easier.

**Comment by nivertius** [comment]
After messing around with code and trying to implement this feature on my own, I decide to wait and see how the XML configuration you mentioned will look like before proceeding, just like you said.

Just as a notice, the `failServerity` *must* be set to `breaking` when using semver-ignore extension, because difference transform changes non-ignored differences to breaking ones (AFAIK). This is what prompted me to create this issue. And new XML configuration should probably take this into account and either force user to set this property to `breaking` or not set it at all, or integrate it with the extension.

**Comment by metlos** [comment]
> Just as a notice, the `failServerity` *must* be set to `breaking` when using semver-ignore extension, because difference transform changes non-ignored differences to breaking ones (AFAIK). This is what prompted me to create this issue. And new XML configuration should probably take this into account and either force user to set this property to `breaking` or not set it at all, or integrate it with the extension.

Yes, semver changes the severity to `breaking` for all differences it finds incompatible with the version increase.

But I don't think `failSeverity` needs to be in sync with this. It doesn't mean "only fail for problems with this severity" but "fail for problems with this severity *or higher*". I.e. `failSeverity=nonBreaking` will fail the build for `nonBreaking`, `potentiallyBreaking` or `breaking` changes. So using semver in conjunction with this shouldn't cause any problem because `breaking` is the highest possible severity. Or have I misunderstood your point?

**Comment by nivertius** [comment]
You are right, `failSeverity` is minimal that breaks build. ;-)

Still, `failSeverity` is meaningless when using semver-ignore and user might think otherwise - that when for example you use semver-ignore and set `failSeverity` to `breaking`, non-breaking API changes will not fail your build when changing patch version number.

That gives me an idea: in semver-ignore, instead of promoting semantic versions conflicts to breaking severity, wouldn't be more intuitive to *just demote* changes that don't produce semantic conflicts to no severity (or just remove them)? This would make above example work as expected.
In other words, why differences have added breaking difference classification in `SemverIgnoreTransform#transform`?
